exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/WebServer.java

```diff
@@ -219,12 +232,43 @@ public void sessionDestroyed(HttpSessionEvent se) {
           securityHandler.logout(sessionAuth);
           session.removeAttribute(SessionAuthentication.__J_AUTHENTICATED);
         }
+
+        // Clear all the custom attributes set as part of session
+        clearSessionCustomAttributes(session);
       }
     });
 
     return new SessionHandler(sessionManager);
   }
 
+  private void clearSessionCustomAttributes(HttpSession session) {
```

Comment:
Moved all the resources inside another class `WebSessionResources` which now manages their lifecycle.